src/os/filestore/FileStore.cc

```diff
@@ -3731,11 +3750,84 @@ int FileStore::_clone_range(const coll_t& cid, const ghobject_t& oldoid, const g
  out:
   lfn_close(o);
  out2:
-  dout(10) << "clone_range " << cid << "/" << oldoid << " -> " << cid << "/" << newoid << " "
+  dout(10) << "clone_range " << oldcid << "/" << oldoid << " -> " << newcid << "/" << newoid << " "
 	   << srcoff << "~" << len << " to " << dstoff << " = " << r << dendl;
   return r;
 }
 
+/*
+ * Move contents of src object according to move_info to base object. Once the move_info is traversed completely, delete the src object.
+ */
+int FileStore::_move_ranges_destroy_src(const coll_t& src_cid, const ghobject_t& src_oid, const coll_t& cid, const ghobject_t& oid,
+                              const vector<boost::tuple<uint64_t, uint64_t, uint64_t>> move_info,
+                              const SequencerPosition& spos)
+{
+  int r = 0;
+  bool err = false;
+  FDRef t, b;
+  int dstcmp, srccmp;
+
+  dout(10) << __func__ << src_cid << "/" << src_oid << " -> " << cid << "/" << oid << dendl;
+
+  // check replay guard for base object. If not possible to replay, return.
+  dstcmp = _check_replay_guard(cid, oid, spos);
+  if (dstcmp < 0)
+    return 0;
+
+  // check the src name too; it might have a newer guard, and we don't
+  // want to clobber it
+  srccmp = _check_replay_guard(src_cid, src_oid, spos);
+  if (srccmp < 0)
+    return 0;
+
+  r = lfn_open(src_cid, src_oid, false, &t);
+  //If we are replaying, it is possible that we do not find src obj as it is deleted before crashing.
+  if (r < 0) {
+    dout(10) << __func__ << " replaying -->" << replaying  << dendl;
+    return 0;
+  }
+
+  r = lfn_open(cid, oid, true, &b);
+  if (r < 0) {
+    lfn_close(t);
+    return 0;
+  }
+
+  for (unsigned i = 0; i < move_info.size(); ++i) {
+     uint64_t srcoff;
+     uint64_t dstoff;
+     uint64_t len;
+
+     srcoff = move_info[i].get<0>();
+     dstoff = move_info[i].get<1>();
+     len = move_info[i].get<2>();
+
+     r = _do_clone_range(**t, **b, srcoff, len, dstoff);
+  }
+
+  dout(10) << __func__  << cid << "/" << oid << " "  <<  " = " << r << dendl;
+
+    lfn_close(b);
+    lfn_close(t);
+
+  //In case crash occurs here, replay will have to do cloning again.
+  //Only if do_clone_range is successful, go ahead with deleting the source object.
+  if (r == 0) {
+    r = lfn_unlink(src_cid, src_oid, spos, true);
+    // If crash occurs between unlink and set guard, correct the error.
+    // as during next time, it might not find the already deleted object.
+    if (r < 0 && replaying) {
+      r = 0;
+    }
+  }
+
+  if (r == 0)
+    _set_replay_guard(cid, spos, false); //set replay guard for base obj coll_t.
```

Comment:
You are checking the replay guard on the object, so you need to set the replay guard on the object.